### PR TITLE
Keras interface and Python 3 compatability

### PR DESCRIPTION
**Pull request by henryre**
Also adds a pretrained CIFAR-10 model

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,7 +5,23 @@
...
 
-#### Hazy Research: [GitHub](https://github.com/HazyResearch), [research homepage](https://snorkel.stanford.edu)
+#### Paper (NIPS 2017): [Learning to Compose Domain-Specific Transformations for Data Augmentation](https://arxiv.org/abs/1709.01643) 
+
+#### Hazy Research: [GitHub](https://github.com/HazyResearch), [research homepage](http://snorkel.stanford.edu)
+
+## NEW: an easy-to-use Keras interface
+
+Just in time for NIPS 2017, we're releasing an easy-to-use
+[Keras](https://keras.io/) interface for TANDA. Just train your TAN, load the
```
Comment (ajratner) [inline README.md]: Re: "Just train your TAN...": I'd say let's lead with a one line (bolded and red?) about Keras interface + pre-trained TAN.  Then mention can be used with any TAN you train / is implied

Comment (ajratner) [reply README.md]: I'd suggest something like: "Just in time for NIPS 2017, we're releasing an easy-to-use, pre-trained substitute for Keras' ImageDataGenerator data augmentation class. Just swap in `TANDAImageDataGenerator` and you'll be using our trained data augmentation generators!"

File: example-scripts/cifar-example.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 source set_env.sh
-python experiments/cifar10/train.py \
+python -m experiments.cifar10.train \
```
Comment (ajratner) [inline example-scripts/cifar-example.sh]: fancy

File: experiments/cifar10/dataset.py [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
+
 from functools import partial
+from six.moves import cPickle
+from skimage import img_as_float
+
 
 def load_cifar10_batch(fpath, one_hot=True, as_float=True):
     with open(fpath, 'rb') as f:
-        data = cPickle.load(f)
+        data = cPickle.load(f, encoding='latin1')
```
Comment (ajratner) [inline experiments/cifar10/dataset.py]: ?

File: keras/tanda_keras.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            zca_whitening=zca_whitening,
+            zca_epsilon=zca_epsilon,
+            rescale=rescale,
+            preprocessing_function=preprocessing_function,
+            data_format=data_format
+        )
+        self.tan = tan
+        self.session = K.get_session()
+
+    def random_transform(self, x, seed=None):
```
Comment (ajratner) [inline keras/tanda_keras.py]: Minor: Update doc string just to clarify that this is not in fact a random transform anymore, just maintaining same interface?

File: keras/tanda_keras.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        data_format: 'channels_first' or 'channels_last'. In 'channels_first'
+            mode, the channels dimension
+            (the depth) is at index 1, in 'channels_last' mode it is at index 3.
+            It defaults to the `image_data_format` value found in your
+            Keras config file at `~/.keras/keras.json`.
+            If you never set it, then it will be "channels_last".
+    """
+
+    def __init__(self,
+                 tan,
```
Comment (ajratner) [inline keras/tanda_keras.py]: Cleaner to just provide the checkpoint path I think?

File: keras/utils.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import json
+
+from experiments.train_scripts import GENERATORS
+from experiments.utils import parse_config_str
+from keras import backend as K
+from tanda.generator import (
```
Comment (ajratner) [inline keras/utils.py]: Is this import necessary?

File: experiments/cifar10/dataset.py (hunk 3 above)
Comment (henryre) [reply experiments/cifar10/dataset.py]: Many stacks overflowed here. Just let it be.

File: keras/tanda_keras.py (hunk 5 above)
Comment (henryre) [reply keras/tanda_keras.py]: Hard disagree

**Comment by ajratner** [comment]
Add a comment linking to said posts?

File: keras/tanda_keras.py (hunk 5 above)
Comment (henryre) [reply keras/tanda_keras.py]: Offline: can either be checkpoint path or TAN object

File: keras/tanda_keras.py (hunk 4 above)
Comment (henryre) [reply keras/tanda_keras.py]: On it

File: keras/utils.py (hunk 6 above)
Comment (henryre) [reply keras/utils.py]: A bit cleaner than redefining IMO

File: keras/tanda_keras.py (hunk 5 above)
Comment (henryre) [reply keras/tanda_keras.py]: Offline: create #4

**Pull request merged**